Phase1/FilterHandler.java

```diff
@@ -0,0 +1,64 @@
+import java.util.*;
+
+public class FilterHandler {
+    private HashSet<String> plus;
+    private HashSet<String> minus;
+    private HashSet<String> noSign;
+    private HashSet<Integer> result;
+    private InvertedIndex invertedIndex;
+    private String[] words;
+
+    public FilterHandler(String[] words, InvertedIndex invertedIndex) {
+        this.words = words;
+        plus = new HashSet<>();
+        minus = new HashSet<>();
+        noSign = new HashSet<>();
+        result = new HashSet<>();
+        this.invertedIndex = invertedIndex;
+    }
+
+    public HashSet<Integer> filter() {
+        addToSeparateSets(words);
+        handleFilters();
+        return result;
+    }
+
+    private void addToSeparateSets(String[] words) {
+        char PLUS_SIGN = '+';
+        char MINUS_SIGN = '-';
+        for (String word : words) {
+            if (word.charAt(0) == PLUS_SIGN)
+                plus.add(word.substring(1, word.length()));
```

Comment:
you can write just plus.add(word.substring(1));
word.length() isn't a necessary parameter.